packages/ember/tests/helpers/link_to_test/link_to_transitioning_classes_test.js

```diff
@@ -1,200 +1,231 @@
 import { RSVP } from 'ember-runtime';
-import { Route, NoneLocation } from 'ember-routing';
-import { run, set } from 'ember-metal';
-import { compile } from 'ember-template-compiler';
-import { Application } from 'ember-application';
-import { jQuery } from 'ember-views';
-import { setTemplates, setTemplate } from 'ember-glimmer';
-
-let Router, App, registry, container;
+import { Route } from 'ember-routing';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
 let aboutDefer, otherDefer;
 
-function bootApplication() {
-  container.lookup('router:main');
-  run(App, 'advanceReadiness');
+function assertHasClass(assert, selector, label) {
+  let testLabel = `${selector.attr('id')} should have class ${label}`;
+
+  assert.equal(selector.hasClass(label), true, testLabel);
 }
 
-function assertHasClass(className) {
-  let i = 1;
-  while (i < arguments.length) {
-    let $a = arguments[i];
-    let shouldHaveClass = arguments[i + 1];
-    equal($a.hasClass(className), shouldHaveClass, $a.attr('id') + ' should ' + (shouldHaveClass ? '' : 'not ') + 'have class ' + className);
-    i += 2;
-  }
+function assertHasNoClass(assert, selector, label) {
+  let testLabel = `${selector.attr('id')} should not have class ${label}`;
+
+  assert.equal(selector.hasClass(label), false, testLabel);
 }
 
 let updateCount, replaceCount;
 
-function sharedSetup() {
-  App = Application.create({
-    name: 'App',
-    rootElement: '#qunit-fixture'
-  });
-
-  App.deferReadiness();
-
-  updateCount = replaceCount = 0;
-  App.Router.reopen({
-    location: NoneLocation.create({
-      setURL(path) {
-        updateCount++;
-        set(this, 'path', path);
-      },
-
-      replaceURL(path) {
-        replaceCount++;
-        set(this, 'path', path);
+moduleFor('The {{link-to}} helper: .transitioning-in .transitioning-out CSS classes', class extends ApplicationTestCase {
+  constructor() {
+    super();
+
+    this.router.map(function() {
+      this.route('about');
+      this.route('other');
+    });
+
+    this.add('route:about', Route.extend({
+      model() {
+        aboutDefer = RSVP.defer();
```

Comment:
Can you swap these various variables to be properties on the test instead of closed over variables? Thank.